rfc-parser/src/main/java/tech/pantheon/yanginator/parser/RFCParser.java

```diff
@@ -10,15 +10,82 @@
 
+import tech.pantheon.yanginator.parser.services.GrammarKitRFCService;
+
 import java.io.File;
 import java.net.URISyntaxException;
-import tech.pantheon.yanginator.parser.services.RFCService;
 
 public class RFCParser {
+    final static GrammarKitRFCService GRAMMARKIT_RFC_SERVICE = new GrammarKitRFCService();
+
     public static void main(String[] args) throws URISyntaxException {
-        RFCService rfcService = new RFCService();
-        File inputFile = rfcService.getFile("main/InputFile.txt");
-        File outputFile = new File("RFCParser/OutputFile");
-        rfcService.getResultFromFile(inputFile, outputFile);
+        for (String arg : args) {
+            System.out.println(arg);
+        }
+
+        generateGrammar(GRAMMARKIT_RFC_SERVICE.getFile("yang-rfc-grammar/shared/rfc-3986.abnf"));
+        generateGrammar("1_0", GRAMMARKIT_RFC_SERVICE.getFile("yang-rfc-grammar/yang-1_0/yang-rfc-6020.abnf"));
+        // TODO Add support for YANG 1.1
+        generateGrammar("1_1", GRAMMARKIT_RFC_SERVICE.getFile("yang-rfc-grammar/yang-1_1/yang-rfc-7950.abnf"));
+    }
+
+    /**
+     * Creates directory named by the version of YANG provided as a parameter, and creates
+     * Backus-Naur form (.bnf) grammar for specified YANG version. The new grammar is created using
+     * <code>myRFCService.transformAbnfToBnf</code>. Any files and directories located inside
+     * gen directory, will be deleted automatically.
+     *
+     * @param yangVersion the name of directory to be created
+     * @param abnfGrammar the file containing .abnf grammar for YANG format
+     */
+    private static void generateGrammar(final String yangVersion, final File abnfGrammar) {
+        String path = "../../main/gen/yang-" + yangVersion;
+        boolean isCreated = new File(path).mkdir();
+        if (isCreated) {
+            File outputFile = new File(path + "/" + abnfGrammar.getName().split("\\.")[0] + "-grammar-kit.bnf");
+            GRAMMARKIT_RFC_SERVICE.transformAbnfToBnf(abnfGrammar, outputFile);
+        } else {
+            File directoryToBeDeleted = new File(path);
+            deleteDirectory(directoryToBeDeleted);
+            generateGrammar(yangVersion, abnfGrammar);
+        }
+    }
+
+    /**
+     * Creates directory named by the file provided as a parameter, and creates
+     * Backus-Naur form (.bnf) grammar for the file. The new grammar is created using
+     * <code>myRFCService.transformAbnfToBnf</code>. Any files and directories located inside
+     * gen directory, will be deleted automatically.
+     *
+     * @param abnfGrammar the file containing .abnf grammar
+     */
+    private static void generateGrammar(final File abnfGrammar) {
+        String path = "../../main/gen/" + abnfGrammar.getName().split("\\.")[0];
+        boolean isCreated = new File(path).mkdir();
+        if (isCreated) {
+            File outputFile = new File(path + "/" + abnfGrammar.getName().split("\\.")[0] + "-grammar-kit.bnf");
```

Comment:
the split() result is used twice:
assign it to a local variable for reuse -- also saves the time to compile the "\\." expression

I guess this part is not that performance-critical, if it were, having a pre-compiled Pattern and going for String.split(Pattern) would be more optimal. That has also the benefit of being more clear with:

   private static final Pattern DOT = Pattern.compile(".", Pattern.LITERAL);

i.e. no need for escapes

Reply:
I have found out, there is another way of retrieving the name of a file.
`String fileName = FilenameUtils.removeExtension(abnfGrammar.getName());`
Would this be fine?

Reply:
Fixed.